File: sketches/sketches.model/src/main/java/edu/kit/kastel/lissa/sketches/model/elements/IBox.java
package edu.kit.kastel.lissa.sketches.model.elements;

import edu.kit.kastel.lissa.sketches.model.impl.Box;
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypes;
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypeMapping;

@SketchBoxTypeMapping(type = SketchBoxTypes.UNKNOWN, implementation = Box.class)
public interface IBox extends ISketchElement {
    SketchBoxTypes getCurrentInterpretation();
}

File: sketches/model/src/main/kotlin/edu/kit/kastel/lissa/sketches/model/elements/IBox.kt
package edu.kit.kastel.lissa.sketches.model.elements

import edu.kit.kastel.lissa.sketches.model.impl.BoxImpl
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypeMapping
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypes
import edu.kit.kastel.lissa.sketches.model.types.SketchElementType

@SketchBoxTypeMapping(type = SketchBoxTypes.UNKNOWN, implementation = BoxImpl::class)
interface IBox : ISketchElement {
    fun currentInterpretation(): SketchBoxTypes
    override fun elementType(): SketchElementType = SketchElementType.BOX
}
